tools/cli/main.cpp

```diff
@@ -19,6 +19,44 @@
 #include <tuple>
 #include <sys/wait.h>
 
+#ifdef __APPLE__
+#include <crt_externs.h>
+#include <cstdlib>
+#include <cerrno>
+#include <cstring>
+#include <unistd.h>
+#define environ (*_NSGetEnviron())
```

Comment:
Avoid using #define to define inline functions in C++, use inline function instead. Believe optimization by compiler

Reply:
done.